ddl/ddl_api.go

```diff
@@ -657,6 +657,31 @@ func checkColumnsAttributes(colDefs []*ast.ColumnDef) error {
 	return nil
 }
 
+func checkTooBigFieldLength(schema *model.DBInfo, colDefs []*ast.ColumnDef, tbInfo *model.TableInfo) error {
+	for _, colDef := range colDefs {
+		if colDef.Tp.Tp == mysql.TypeVarchar {
+			cs := tbInfo.Charset
+			if len(tbInfo.Charset) == 0 {
+				if len(schema.Charset) == 0 {
+					cs = mysql.DefaultCharset
+				} else {
+					cs = schema.Charset
+				}
+			}
+			desc, err := charset.GetCharsetDesc(cs)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			maxFlen := mysql.MaxFieldVarCharLength
```

Comment:
Use common function to extract the logic, don't write redundant code.

Reply:
Done.